### PR TITLE
create internet gateway resource

### PR DESCRIPTION
**Pull request by haim-kermany**
added internet gateway as a resource
creating it during parsing flow.
next step will be to support public subnets

### CODE REVIEW

File: pkg/awsvpc/parser.go [hunk 1]
```diff
@@ -291,6 +300,49 @@ func (rc *AWSresourcesContainer) getSGconfig(
...
+		}
+		vpc := res.GetVPC(vpcUID).(*commonvpc.VPC)
+		subnets := vpc.Subnets()
+		// todo - use only the public subnet:
+		if len(subnets) == 0 {
+			logging.Warnf("skipping internet gateway %s - it does not have any attached subnet\n", igwName)
+			continue
+		}
+		routerIgw := newIGW(igwName, igwID, subnets, vpc)
+		// TODO - where to put this resource?
```
Comment (adisos) [inline pkg/awsvpc/parser.go:325]: what does this TODO mean?

File: pkg/awsvpc/parser.go [hunk 2]
```diff
@@ -291,6 +300,49 @@ func (rc *AWSresourcesContainer) getSGconfig(
...
+			logging.Warnf("skipping internet gateway %s - it has %d vpcs attached\n", igwName, len(igw.Attachments))
+			continue
+		}
+		vpcUID := *igw.Attachments[0].VpcId
+		if skipByVPC[vpcUID] {
+			continue
+		}
+		vpc := res.GetVPC(vpcUID).(*commonvpc.VPC)
+		subnets := vpc.Subnets()
+		// todo - use only the public subnet:
```
Comment (adisos) [inline pkg/awsvpc/parser.go:319]: is this planned to a follow up PR?

**Review by adisos — COMMENTED** [review]
looks good, added few small questions

File: pkg/awsvpc/parser.go (hunk 1 above)
Comment (haim-kermany) [reply pkg/awsvpc/parser.go:325]: removed the todo comment 
(it was not clear if the itg was per region or per vpc, )

File: pkg/awsvpc/parser.go (hunk 2 above)
Comment (haim-kermany) [reply pkg/awsvpc/parser.go:319]: yes

**Review by adisos — APPROVED** [review]

**Pull request merged**